src/ai/backend/common/docker.py

```diff
@@ -398,83 +341,163 @@ def __eq__(self, other):
         return self._data == other
 
 
+class ParsedImageStr(namedtuple("ParsedImageStr", ["registry", "project_and_image_name", "tag"])):
```

Comment:
Just a comment.
How about using [typing.NamedTuple](https://docs.python.org/3/library/typing.html#typing.NamedTuple)?
`collections.namedtuple` does not support type hint of attributes. In this case, we cannot set type hint for `registry`, `project_and_image_name` and `tag`